pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2730,11 +2840,26 @@ func (c *Cloud) EnsureLoadBalancerDeleted(clusterName string, service *api.Servi
 
 		// Collect the security groups to delete
 		securityGroupIDs := map[string]struct{}{}
+
+		// shared securitygroup ID for reference
+		var ssgID string
+		var err error
+		if s.cfg.Global.EnableSharedSecurityGroupIngress {
+			ssgID, err = s.ensureSharedSecurityGroup()
+			if err != nil {
+				glog.Warningf("Error creating shared security group: ", err)
```

Comment:
Probably should be a hard-error - i.e. return fmt.Errorf.  Because we don't want to delete the shared security group if the AWS API has a transient failure here.  Although I guess that the shared security deletion will fail because it is attached to other ELBs?